net/ntske/provider.go

```diff
@@ -0,0 +1,105 @@
+package ntske
+
+import (
+	"crypto/rand"
+	"math"
+	"sync"
+	"time"
+)
+
+/*
+This provider is set up to be used concurrently by the NTSKE and time servers.
+In case they should not run on the same machine one option would be to synchronize
+an initial key once at startup of the servers and then each of them will separately
+create the next key each day using some key derivation function like hkdf.
+*/
+
+const (
+	keyValidity        time.Duration = time.Hour * 24 * 3
+	keyRenewalInterval time.Duration = time.Hour * 24
+)
+
+type Key struct {
+	Id       int
+	Value    []byte
+	Validity struct {
+		NotBefore time.Time
+		NotAfter  time.Time
+	}
+}
+
+type Provider struct {
+	mu          sync.Mutex
+	keys        map[int]Key
+	currentID   int
+	generatedAt time.Time
+}
+
+func (k *Key) IsValid() bool {
```

Comment:
I'm still not happy about the implicit time dependence of this function. How about maiking it explicit like, e.g.,
```
func (k *Key) IsValidAt(t time.Time) bool {
	return !t.Before(k.Validity.NotBefore) && !t.After(k.Validity.NotAfter)
}
```
We would then also need to call `time.Now()` only once at the top of `generateNext`.

Reply:
Done